include/circt/Dialect/Ibis/IbisOps.td

```diff
@@ -145,4 +160,110 @@ def CallOp : IbisOp<"call", [CallOpInterface]> {
   }];
 }
 
+
+// ===---------------------------------------------------------------------===//
+// Low-level Ibis operations
+// ===---------------------------------------------------------------------===//
+
+def ContainerOp : IbisOp<"container", [
+  IsolatedFromAbove, RegionKindInterface,
+  Symbol, SymbolTable, SingleBlock,
+  NoTerminator, NoRegionArguments,
+  HasParent<"ClassOp">
+]> {
+  let summary = "Ibis container";
+  let description = [{
+    An ibis container describes a collection of logic nested within an Ibis class.
+  }];
+
+  let arguments = (ins SymbolNameAttr:$sym_name);
+  let regions = (region SizedRegion<1>:$body);
+
+  let assemblyFormat = [{
+    $sym_name attr-dict-with-keyword $body
+  }];
+
+  let extraClassDeclaration = [{
+    // Implement RegionKindInterface.
+    static RegionKind getRegionKind(unsigned index) { return RegionKind::Graph; }
```

Comment:
I don't know that this needs to be a graph region... @blakep-msft ?

Reply:
I suppose it doesn't really matter.  There will not be operations that "execute" inside of a container, just scheduled pipelines and operations to access ports.